src/MLJDecisionTreeInterface.jl

```diff
@@ -129,13 +129,66 @@ function MMI.fit(
                              m.min_samples_split,
                              m.min_purity_increase;
                              rng=m.rng)
-    cache  = nothing
+    cache  = deepcopy(m)
 
     report = (features=features,)
 
     return (forest, classes_seen, integers_seen), cache, report
 end
 
+function MMI.update(
+    model::RandomForestClassifier,
+    verbosity::Int,
+    old_fitresult,
+    old_model,
+    Xmatrix,
+    yplain,
+    features,
+    classes,
+    )
+
+    only_iterations_have_changed = MMI.is_same_except(model, old_model, :n_trees)
+
+    if !only_iterations_have_changed
+        return MMI.fit(
+            model,
+            verbosity,
+            Xmatrix,
+            yplain,
+            features,
+            classes,
+        )
+    end
+
+    old_forest = old_fitresult[1]
+    Δn_trees = model.n_trees - old_model.n_trees
+    # if `n_trees` drops, then tuncate, otherwise compute more trees
+    if Δn_trees < 0
+        verbosity > 0 && @info "Dropping $(-Δn_trees) trees from the forest. "
+        forest = old_forest[1:model.n_trees]
+    else
+        verbosity > 0 && @info "Adding $Δn_trees trees to the forest. "
+        forest = DT.build_forest(
+            old_forest,
+            yplain, Xmatrix,
+            model.n_subfeatures,
+            model.n_trees,
+            model.sampling_fraction,
+            model.max_depth,
+            model.min_samples_leaf,
+            model.min_samples_split,
+            model.min_purity_increase;
+            rng=model.rng,
+        )
+    end
+
+    fitresult = (forest, old_fitresult[2:3]...)
+    cache = deepcopy(model)
```

Comment:
Should the cache be a tuple? `cache = (deepcopy(model), )`

Reply:
No, there's nothing in the spec about the form of `cache`.